module/zfs/dbuf.c

```diff
@@ -3529,7 +3529,7 @@ dbuf_write_children_ready(zio_t *zio, arc_buf_t *buf, void *vdb)
 	dmu_buf_impl_t *db = vdb;
 	dnode_t *dn;
 	blkptr_t *bp;
-	uint64_t i;
+	int i;
```

Comment:
Casting the `1` in the for loop below to `1ULL` would probably be a better way to handle this.